### DISCUSSION TITLE
Make the random button less random

### DISCUSSION THREAD

**Discussion by EpiCreeper** [Quaver/Quaver#4806] [Feature Request]
### Is your feature request related to a problem? Please describe.

If you blindly pull 6 balls from a container of red and blue marbles, getting red, blue, blue, red, red, blue, is just as likely as pulling blue, blue, blue, red, red, red. This might be the case but we humans think the second option is less random because it has less variety. The same goes for the random button in song select. If I press refresh and get the same map 3 times out of 20 it feels less random even though it is truly randomized.

### Describe the solution you'd like

After pressing f2, the game would save each map id in a list and would know not to pick it again after pressing the random button. There could also be a slight bias against maps with the same artist or mapper to make it feel even more random.

### Describe alternatives you've considered

People might want the button to stay truly random so it could be a good idea to keep it as an option in the settings.

### Additional context

Making the random button less random might sound counter intuitive, but its actually something most music streaming services do already.

**Comment by A6sT** [comment]
using weight based randomness with recent random mapset selection having lower base-weight sounds like the way to go
